Gems/EMotionFX/Code/Source/Editor/Plugins/Ragdoll/PhysicsSetupColliderTranslationManipulators.cpp

```diff
@@ -0,0 +1,200 @@
+            [this](const AzToolsFramework::SurfaceManipulator::Action& action)
+            {
+                BeginEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        // mouse move callbacks
+        m_translationManipulators.InstallLinearManipulatorMouseMoveCallback(
+            [this](const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                OnManipulatorMoved(action.m_start.m_localPosition, action.m_current.m_localPositionOffset);
+            });
+
+        m_translationManipulators.InstallPlanarManipulatorMouseMoveCallback(
+            [this](const AzToolsFramework::PlanarManipulator::Action& action)
+            {
+                OnManipulatorMoved(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        m_translationManipulators.InstallSurfaceManipulatorMouseMoveCallback(
+            [this](const AzToolsFramework::SurfaceManipulator::Action& action)
+            {
+                OnManipulatorMoved(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        // mouse up callbacks
+        m_translationManipulators.InstallLinearManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::LinearManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localPositionOffset);
+            });
+
+        m_translationManipulators.InstallPlanarManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::PlanarManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+
+        m_translationManipulators.InstallSurfaceManipulatorMouseUpCallback(
+            [this](const AzToolsFramework::SurfaceManipulator::Action& action)
+            {
+                FinishEditing(action.m_start.m_localPosition, action.m_current.m_localOffset);
+            });
+    }
+
+    void PhysicsSetupColliderTranslationManipulators::Refresh()
+    {
+        if (!m_physicsSetupManipulatorData.m_valid || !m_physicsSetupManipulatorData.m_colliderNodeConfiguration ||
+            m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes.empty())
+        {
+            return;
+        }
+
+        m_translationManipulators.SetLocalPosition(
+            m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes[0].first->m_position);
+    }
+
+    void PhysicsSetupColliderTranslationManipulators::Teardown()
+    {
+        m_translationManipulators.Unregister();
+    }
+
+    void PhysicsSetupColliderTranslationManipulators::ResetValues()
+    {
+        if (!m_physicsSetupManipulatorData.m_valid || !m_physicsSetupManipulatorData.m_colliderNodeConfiguration ||
+            m_physicsSetupManipulatorData.m_colliderNodeConfiguration->m_shapes.empty())
+        {
```

Comment:
nit: this check appears 4 times in the file, maybe it's worth moving it to a function.

Reply:
Yeah, that's a good idea. I'm still going to be touching this code in the next PR, so I'll fix it then.